osutil/mountentry_linux.go

```diff
@@ -26,6 +26,46 @@ import (
 	"syscall"
 )
 
+// This is needed to properly split options that could contain an escaped
+// ',' character.
+func SplitMountOptions(s string) []string {
+	var o []string
+
+	var cur strings.Builder
+	escaped := false
+	for _, c := range s {
+		switch c {
+		case '\\':
+			if escaped {
+				cur.WriteRune(c)
+				escaped = false
+			} else {
+				escaped = true
+			}
+		case ',':
+			if escaped {
+				cur.WriteRune(c)
+				escaped = false
+			} else {
+				o = append(o, cur.String())
+				cur.Reset()
+			}
+		default:
+			if escaped {
+				cur.WriteRune('\\')
```

Comment:
Looking at the kernel, this is not how the parsing algorithm works for overlayfs. `\f` is parsed as `f`. Not as `\f`.

Reply:
I reverted the logic to the old one for `ParseMountEntry`.